pkg/connector/helpers.go

```diff
@@ -50,7 +51,16 @@ func (o *orgNameCache) GetOrgName(ctx context.Context, orgID *v2.ResourceId) (st
 		return "", err
 	}
 
-	org, _, err := o.c.Organizations.GetByID(ctx, oID)
+	client := o.c
+	if len(o.appClients) != 0 {
+		var ok bool
+		client, ok = o.appClients[oID]
+		if !ok {
+			return "", fmt.Errorf("organization: %d doesn't exist", oID)
+		}
+	}
```

Comment:
I'm not sure if its okay to cache like this, we generally don't rely on state like this, I think there can be times when we resume a sync